java/fury-core/src/main/java/org/apache/fury/serializer/collection/ChildContainerSerializers.java

```diff
@@ -151,11 +154,14 @@ public Collection newCollection(MemoryBuffer buffer) {
     }
 
     @Override
-    public T copy(T originCollection) {
-      if (objectSerializer == null) {
-        objectSerializer = new ObjectSerializer<>(fury, type, false);
+    public Collection newCollection(Collection originCollection) {
+      Collection newCollection = super.newCollection(originCollection);
+      if (fieldInfos == null) {
+        List<Field> fields = ReflectionUtils.getFieldsWithoutSuperClasses(type, superClasses);
```

Comment:
We can just `fury.getClassResolver().getAllDescriptorsMap(beanClass, false).values(); ` or `org.apache.fury.reflect.ReflectionUtils#getFields` instead